aws_lambda_powertools/utilities/idempotency/persistence/dynamodb.py

```diff
@@ -15,8 +15,9 @@
 from aws_lambda_powertools.utilities.idempotency.exceptions import (
     IdempotencyItemAlreadyExistsError,
     IdempotencyItemNotFoundError,
+    IdempotencyValidationError,
 )
-from aws_lambda_powertools.utilities.idempotency.persistence.base import (
+from aws_lambda_powertools.utilities.idempotency.persistence.datarecord import (
```

Comment:
Just picking this example to demonstrate how this change is a huge breaking change. If someone has implemented its own persistence store, the code will now break. Can we keep compatibility?

Reply:
Great point, Ruben!! But we are i[mporting DataRecord](https://github.com/aws-powertools/powertools-lambda-python/pull/3446/files#diff-becc2a8b1233bddbb0d38b13a8634ab96bdd5d00f16f0514f6ea1df509c3b1d6R26) in `aws_lambda_powertools.utilities.idempotency.persistence.base` and customers can continue importing in the same way, this will not break this import. Pls let me know if I'm missing something here.

```python
from aws_lambda_powertools.utilities.idempotency.persistence.base import DataRecord as DR1 # OLD IMPORT
from aws_lambda_powertools.utilities.idempotency.persistence.datarecord import DataRecord as DR2 # NEW IMPORT

dr1 = DR1("idkey","pending",1,1,"response", "payload") # OLD IMPORT

dr2 = DR2("idkey","pending",1,1,"response", "payloadxxxxxxxxxxx") # NEW IMPORT

# Values are different, this must fail
assert dr1.__dict__ == dr2.__dict__, "AssertionError - Values are different"
```

```sh
❯ python test.py
Traceback (most recent call last):
  File "/tmp/ddbcondition/test.py", line 9, in <module>
    assert dr1.__dict__ == dr2.__dict__, "AssertionError - Values are different"
AssertionError: AssertionError - Values are different
```